adobe/aem-core-cif-components
# CIF-961 - Shopping cart (MiniCart) component using GraphQL API

This PR is huge because it contains all the React components needed to render the MiniCart. 

## Description

The MiniCart is a React app which renders in the `minicart` placeholder div. The component uses React hooks provided by Apollo to fetch/mutate cart data.
The "Cart Trigger" icon has been converted to a React component as well because it was easier to interact with it (change the style when there are items in cart and increase de cart counter). The "Cart Trigger" is rendered in the header using [React Portals ](https://reactjs.org/docs/portals.html).
The `Add to cart` button has been updated to work with the new `minicart` component.

### How to test it on a local instance

Prerequisites:
* Make sure you have a CIF project using the latest release of `aem-core-cif-components`. 
* Make sure you have a Mangento 2.3.2 instance running on HTTPS
* Make sure you have your dispatcher set up
* (Optional) Make sure you have the `repo` command installed so that the build process can push the clientlib to AEM

Once the above requirements are met go to `ui.apps/src/main/javascript/minicart` and run 
```
yarn install && yarn build:clientlib
```
The `build:clientlib` command will copy the files from the `./dist` folder to the clientlib and also run the `repo` command to upload them to AEM. If you don't have `repo` utility installed then you can just copy the `./dist` folder in `./ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/minicart/v1/minicart/clientlib` and then install the components content-package.

### Leftovers

There are still some minor things to do:
- the styles are not 100% on par with Venia, especially the loading states.
- there are no tests yet

There are also some things that could be done better, such as leveraging React Context instead of just passing function handlers along a (quite deep) tree of components.

## Related Issue

CIF-961

## Motivation and Context

Align with the technology used by Magento PWA 

## How Has This Been Tested?

Functional testing

## Screenshots (if appropriate):

## Types of changes


## Checklist:
